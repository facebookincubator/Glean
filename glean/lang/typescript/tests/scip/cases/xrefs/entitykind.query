query: lsif.EntityKind { entity = { typescript = _ } } 
transform: [gensort, []]
