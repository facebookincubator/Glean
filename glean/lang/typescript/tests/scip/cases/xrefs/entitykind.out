[
  "@generated",
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 14,
                        "columnBegin": 7,
                        "lineEnd": 14,
                        "columnEnd": 9
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/Git#" },
                  "ident": { "key": "`example.ts`/Git#" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 15,
                        "columnBegin": 23,
                        "lineEnd": 15,
                        "columnEnd": 25
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": {
                    "key": "`example.ts`/Git#`<constructor>`().(dir)"
                  },
                  "ident": { "key": "`example.ts`/Git#`<constructor>`().(dir)" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 16,
                        "columnBegin": 11,
                        "lineEnd": 16,
                        "columnEnd": 13
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              }
            }
          }
        }
      },
      "kind": 1
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 16,
                        "columnBegin": 41,
                        "lineEnd": 16,
                        "columnEnd": 43
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/cwd0:" },
                  "ident": { "key": "`example.ts`/cwd0:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 16,
                        "columnBegin": 51,
                        "lineEnd": 16,
                        "columnEnd": 56
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/silent0:" },
                  "ident": { "key": "`example.ts`/silent0:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 24,
                        "columnBegin": 7,
                        "lineEnd": 24,
                        "columnEnd": 9
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/cwd1:" },
                  "ident": { "key": "`example.ts`/cwd1:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 25,
                        "columnBegin": 7,
                        "lineEnd": 25,
                        "columnEnd": 12
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/silent1:" },
                  "ident": { "key": "`example.ts`/silent1:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 27,
                        "columnBegin": 48,
                        "lineEnd": 27,
                        "columnEnd": 50
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/cwd2:" },
                  "ident": { "key": "`example.ts`/cwd2:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 27,
                        "columnBegin": 58,
                        "lineEnd": 27,
                        "columnEnd": 63
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/silent2:" },
                  "ident": { "key": "`example.ts`/silent2:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 30,
                        "columnBegin": 7,
                        "lineEnd": 30,
                        "columnEnd": 9
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/cwd3:" },
                  "ident": { "key": "`example.ts`/cwd3:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 31,
                        "columnBegin": 7,
                        "lineEnd": 31,
                        "columnEnd": 12
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/silent3:" },
                  "ident": { "key": "`example.ts`/silent3:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 34,
                        "columnBegin": 3,
                        "lineEnd": 34,
                        "columnEnd": 8
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/Git#commit()." },
                  "ident": { "key": "`example.ts`/Git#commit()." }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 34,
                        "columnBegin": 10,
                        "lineEnd": 34,
                        "columnEnd": 12
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/Git#commit().(msg)" },
                  "ident": { "key": "`example.ts`/Git#commit().(msg)" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 34,
                        "columnBegin": 23,
                        "lineEnd": 34,
                        "columnEnd": 26
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/Git#commit().(date)" },
                  "ident": { "key": "`example.ts`/Git#commit().(date)" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 34,
                        "columnBegin": 37,
                        "lineEnd": 34,
                        "columnEnd": 42
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/Git#commit().(author)" },
                  "ident": { "key": "`example.ts`/Git#commit().(author)" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 35,
                        "columnBegin": 11,
                        "lineEnd": 35,
                        "columnEnd": 16
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              }
            }
          }
        }
      },
      "kind": 1
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 35,
                        "columnBegin": 45,
                        "lineEnd": 35,
                        "columnEnd": 47
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/cwd4:" },
                  "ident": { "key": "`example.ts`/cwd4:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 35,
                        "columnBegin": 60,
                        "lineEnd": 35,
                        "columnEnd": 65
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/silent4:" },
                  "ident": { "key": "`example.ts`/silent4:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 46,
                        "columnBegin": 17,
                        "lineEnd": 46,
                        "columnEnd": 30
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/createTempRepo()." },
                  "ident": { "key": "`example.ts`/createTempRepo()." }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 46,
                        "columnBegin": 36,
                        "lineEnd": 46,
                        "columnEnd": 42
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              }
            }
          }
        }
      },
      "kind": 1
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 46,
                        "columnBegin": 53,
                        "lineEnd": 46,
                        "columnEnd": 55
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              }
            }
          }
        }
      },
      "kind": 1
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 47,
                        "columnBegin": 9,
                        "lineEnd": 47,
                        "columnEnd": 15
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              }
            }
          }
        }
      },
      "kind": 1
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 49,
                        "columnBegin": 9,
                        "lineEnd": 49,
                        "columnEnd": 11
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              }
            }
          }
        }
      },
      "kind": 1
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 51,
                        "columnBegin": 11,
                        "lineEnd": 51,
                        "columnEnd": 17
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/repoDir0:" },
                  "ident": { "key": "`example.ts`/repoDir0:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  },
  {
    "key": {
      "entity": {
        "typescript": {
          "defn": {
            "key": {
              "defn": {
                "key": {
                  "file": {
                    "key": { "file": { "key": "example.ts" }, "language": 49 }
                  },
                  "range": {
                    "key": {
                      "range": {
                        "lineBegin": 51,
                        "columnBegin": 20,
                        "lineEnd": 51,
                        "columnEnd": 22
                      },
                      "text": { "key": "" }
                    }
                  }
                }
              },
              "moniker": {
                "key": {
                  "kind": 0,
                  "scheme": { "key": "`example.ts`/git0:" },
                  "ident": { "key": "`example.ts`/git0:" }
                }
              }
            }
          }
        }
      },
      "kind": 26
    }
  }
]